app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt

```diff
@@ -109,13 +111,19 @@ class SelfUserProfileViewModel @Inject constructor(
     init {
         viewModelScope.launch {
             fetchSelfUser()
+            checkIfUserAbleToMigrateToTeamAccount()
             observeEstablishedCall()
             fetchIsReadOnlyAccount()
             observeLegalHoldStatus()
             markCreateTeamNoticeAsRead()
         }
     }
 
+    private suspend fun checkIfUserAbleToMigrateToTeamAccount() {
+        val isAbleToMigrateToTeamAccount = isPersonalToTeamAccountSupportedByBackend() && userProfileState.teamName.isNullOrBlank()
+        userProfileState = userProfileState.copy(isAbleToMigrateToTeamAccount = isAbleToMigrateToTeamAccount)
```

Comment:
This could all be in the Kalium UseCase instead of in the ViewModel.

The ViewModel ideally shouldn't know about whether a backend needs to support something or not.

It could just be `ShouldOfferPersonalToTeamMigrationUseCase`, and all the logic could be in there.

ViewModel just gets a true/false and that's it.